src/lib/containers/access_requirement_list/AcceptedRequirements.tsx

```diff
@@ -0,0 +1,179 @@
+import * as React from 'react'
+import { useState, useEffect } from 'react'
+import MarkdownSynapse from '../../../lib/containers/MarkdownSynapse'
+import {
+  WikiPageKey,
+  AccessRequirementStatus,
+  AccessRequirement,
+  UserProfile,
+  TermsOfUseAccessRequirement,
+  SelfSignAccessRequirement,
+  ManagedACTAccessRequirement,
+  AccessApproval,
+  ApprovalState,
+  ACTAccessRequirement,
+} from '../../../lib/utils/synapseTypes/'
+import { SynapseClient } from '../../../lib/utils'
+import AccessApprovalCheckMark from './AccessApprovalCheckMark'
+import { SUPPORTED_ACCESS_REQUIREMENTS } from './AccessRequirementList'
+
+export type AcceptedRequirementsProps = {
+  user: UserProfile | undefined
+  token: string | undefined
+  wikiPage: WikiPageKey | undefined
+  accessRequirement:
+    | AccessRequirement
+    | TermsOfUseAccessRequirement
+    | SelfSignAccessRequirement
+    | ManagedACTAccessRequirement
+  accessRequirementText: string
+  accessRequirementStatus: AccessRequirementStatus | undefined
+  showButton?: boolean
+  onHide?: Function
+}
+
+export default function AcceptedRequirements({
+  user,
+  token,
+  wikiPage,
+  accessRequirement,
+  accessRequirementText,
+  accessRequirementStatus,
+  showButton,
+  onHide,
+}: AcceptedRequirementsProps) {
+  const [isHide, setIsHide] = useState<boolean>(true)
+  const propsIsApproved = accessRequirementStatus?.isApproved
+  const [isApproved, setIsApproved] = useState<boolean | undefined>(
+    propsIsApproved,
+  )
+  const [buttonText, setButtonText] = useState<string>()
+
+  useEffect(() => {
+    const setIsApprovedValueFromProps = (propsIsApproved?: boolean) => {
+      setIsApproved(propsIsApproved)
+    }
+    setIsApprovedValueFromProps(propsIsApproved)
+
+    if (
+      accessRequirement.concreteType ===
+      SUPPORTED_ACCESS_REQUIREMENTS.ManagedACTAccessRequirement
+    ) {
+      setButtonText(`Get ${accessRequirementText} via synapse.org`)
```

Comment:
I would recommend removing this as a state variable and instead making it a computed value at render